bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -3747,57 +3746,29 @@ void ReplicaImp::executeRequestsInPrePrepareMsg(concordUtils::SpanWrapper &paren
       LOG_DEBUG(CNSUS, "Consensus reached");
     }
     SCOPED_MDC("pp_msg_cid", ppMsg->getCid());
+    std::deque<IRequestsHandler::ExecutionRequest> accumulatedRequests;
     while (reqIter.getAndGoToNext(requestBody)) {
       size_t tmp = reqIdx;
       reqIdx++;
       if (!requestSet.get(tmp)) {
         continue;
       }
-
       ClientRequestMsg req((ClientRequestMsgHeader *)requestBody);
       if (req.flags() & EMPTY_CLIENT_FLAG) {
         continue;
       }
       SCOPED_MDC_CID(req.getCid());
       NodeIdType clientId = req.clientProxyId();
-
-      uint32_t actualReplyLength = 0;
-      uint32_t actualReplicaSpecificInfoLength = 0;
-      int status;
-      {
-        TimeRecorder scoped_timer(*histograms_.executeWriteRequest);
-        status = bftRequestsHandler_.execute(
-            clientId,
-            lastExecutedSeqNum + 1,
-            req.flags(),
-            req.requestLength(),
-            req.requestBuf(),
-            ReplicaConfig::instance().getmaxReplyMessageSize() - sizeof(ClientReplyMsgHeader),
-            replyBuffer,
-            actualReplyLength,
-            actualReplicaSpecificInfoLength,
-            span);
-      }
-
-      ConcordAssertGT(actualReplyLength,
-                      0);  // TODO(GG): TBD - how do we want to support empty replies? (actualReplyLength==0)
-
-      if (status != 0) {
-        const auto requestSeqNum = req.requestSeqNum();
-        LOG_WARN(CNSUS, "Request execution failed: " << KVLOG(clientId, requestSeqNum));
-      } else {
-        if (req.flags() & HAS_PRE_PROCESSED_FLAG) metric_total_preexec_requests_executed_.Get().Inc();
-        ClientReplyMsg *replyMsg = clientsManager->allocateNewReplyMsgAndWriteToStorage(
-            clientId, req.requestSeqNum(), currentPrimary(), replyBuffer, actualReplyLength);
-        replyMsg->setReplicaSpecificInfoLength(actualReplicaSpecificInfoLength);
-        send(replyMsg, clientId);
-        delete replyMsg;
-      }
-
-      clientsManager->removePendingRequestOfClient(clientId);
+      accumulatedRequests.push_back(IRequestsHandler::ExecutionRequest{
```

Comment:
I think this could be put in executeRequestsAndSendResponses, too, as it is not used in this function anymore.

Reply:
Moved all the part of accumulating to executeRequestsAndSendResponses.